packages/annotorious/src/annotation/tools/drawingToolsRegistry.ts

```diff
@@ -1,9 +1,9 @@
-import type { SvelteComponent } from 'svelte';
-import { RubberbandRectangle } from './rectangle';
-import { RubberbandPolygon } from './polygon';
+import RubberbandRectangle from './rectangle/RubberbandRectangle.svelte';
+import RubberbandPolygon from './polygon/RubberbandPolygon.svelte';
 import type { DrawingMode } from '../../AnnotoriousOpts';
+import type { ShapeName } from 'src/model';
 
-export type DrawingTool = 'rectangle' | 'polygon' | string;
+export type DrawingTool = 'RECTANGLE' | 'POLYGON';
```

Comment:
This is something we'd need to keep the same, in order not to break the consuming apps. The `| string` isn't ideal either I admit. But the basic idea is that plugins are going to register new tools (with new names), so I don't see a way to keep this strongly typed without breaking extensibility.

Reply:
Hmm I think in that case it should probably just be string. I changed this because there are some benefits to having the tool be the same as the shape name, but on reflection that probably is just wrong -- no reason to assume that there is only one way to edit polygon shapes.

Reply:
> no reason to assume that there is only one way to edit polygon shapes.

My thoughts exactly when designing this: you can have different tools that produce the same shape type. (Annotorious v2 already did.) I agree it could just be string. Originally, I added `rectangle` and `polygon` to get a bit of IDE auto-completion, although it doesn't actually work. (For odd reasons, it would have to be `| string & {};` but I never actually added this...)